stratum/hal/lib/tdi/dpdk/dpdk_hal.cc

```diff
@@ -207,14 +210,45 @@ ::util::Status DpdkHal::Run() {
   const std::vector<std::string> external_stratum_urls =
       absl::StrSplit(FLAGS_external_stratum_urls, ',');
   {
+    std::string log_output_str;
     ::grpc::ServerBuilder builder;
     SetGrpcServerKeepAliveArgs(&builder);
 
-    builder.AddListeningPort(FLAGS_local_stratum_url,
-                             ::grpc::InsecureServerCredentials());
-
-    for (const auto& url : external_stratum_urls) {
-      builder.AddListeningPort(url, ::grpc::InsecureServerCredentials());
+    if (FLAGS_grpc_open_insecure_mode) {
```

Comment:
Shouldn't missing certificates trigger defaulting to insecure mode? Has PSE asked to error out if certificates don't exist.

Reply:
No, we dont want to default to insecure mode (what it used to be in the past) if certificates are missing. 

We are providing the user to launch the process in an insecure_mode via a runtime flag, but it was a conscious decision taken by the user. We dont want the user to think they are using secure mode communication while the underlying sockets are unsecured.